django/db/models/fields/related.py

```diff
@@ -2630,6 +2630,14 @@ def contribute_to_class(self, cls, name, **kwargs):
         # clash.
         if self.rel.symmetrical and (self.rel.to == "self" or self.rel.to == cls._meta.object_name):
             self.rel.related_name = "%s_rel_+" % name
+        else:
+            # If the backwards relation is disabled, replace the original
+            # related_name with one generated from the m2m field name.
+            # This is because Django still uses backwards relations internally
+            # and we need to avoid clashes (multiple m2m fields with
+            # related_name == '+').
+            if self.rel.is_hidden():
```

Comment:
Should we check explicitly only for name '+'. I'm not sure if users can define some other hidden related name if they want to. In that case we shouldn't override the user's hidden name with an auto generated name.